### PR TITLE
Igor lig 503 pip assert embeddings file in correct

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

- add checks whether the embeddings follow the specs
- add test case
- fix a small bug where we use a wrong header for embeddings in the API workflow test (the bug had no impact)

### CODE REVIEW

**Review by japrescott — APPROVED** [review]

File: tests/utils/test_io.py [hunk 1]
```diff
@@ -33,3 +36,32 @@ def test_valid_filenames(self):
...
+class TestEmbeddingsIO(unittest.TestCase):
+
+    def test_valid_embeddings(self):
+        # correct embedding file as created through lightly
+        embeddings_path = tempfile.mktemp('.csv', 'embeddings')
+        embeddings = np.random.rand(32, 2)
+        labels = [0 for i in range(len(embeddings))]
+        filenames = [f'img_{i}.jpg' for i in range(len(embeddings))]
+        save_embeddings(embeddings_path, embeddings, labels, filenames)
+        check_embeddings(embeddings_path)
```
Comment (MalteEbner) [inline tests/utils/test_io.py:49]: Please put this in a separate subtest
`with self.subTest("valid embeddings file"):`

File: tests/utils/test_io.py [hunk 2]
```diff
@@ -33,3 +36,32 @@ def test_valid_filenames(self):
...
+
+        # should fail because there whitespaces in the header columns
+        lines = ['filenames, embedding_0,embedding_1,labels\n',
+                 'img_1.jpg, 0.351,0.1231']
+        with open(embeddings_path, 'w') as f:
+            f.writelines(lines)
+        with self.assertRaises(RuntimeError) as context:
```
Comment (MalteEbner) [inline tests/utils/test_io.py:56]: Please put this in a separate subtest
with self.subTest("whitespace embeddings file"):

File: tests/utils/test_io.py [hunk 3]
```diff
@@ -33,3 +36,32 @@ def test_valid_filenames(self):
...
+            check_embeddings(embeddings_path)
+        self.assertTrue('must not contain whitespaces' in str(context.exception))
+
+        # should fail because there is no `labels` column in the header
+        lines = ['filenames,embedding_0,embedding_1\n',
+                 'img_1.jpg,0.351,0.1231']
+        with open(embeddings_path, 'w') as f:
+            f.writelines(lines)
+        with self.assertRaises(RuntimeError) as context:
```
Comment (MalteEbner) [inline tests/utils/test_io.py:65]: Please put this in a separate subtest
with self.subTest("no label in embeddings file"):

**Review by MalteEbner — COMMENTED** [review]
The function looks good to me. Great that you added unittests! However, please put them in separate subtests, so that it is clearer which of them fails if they fail.

File: tests/utils/test_io.py [hunk 4]
```diff
@@ -33,3 +36,36 @@ def test_valid_filenames(self):
...
+        filenames = [f'img_{i}.jpg' for i in range(len(embeddings))]
+        save_embeddings(self.embeddings_path, embeddings, labels, filenames)
+
+    def test_valid_embeddings(self):
+        check_embeddings(self.embeddings_path)
+
+    def test_whitespace_in_embeddings(self):
+        # should fail because there whitespaces in the header columns
+        lines = ['filenames, embedding_0,embedding_1,labels\n',
+                 'img_1.jpg, 0.351,0.1231']
```
Comment (MalteEbner) [inline tests/utils/test_io.py:56]: I am not sure why you add the 2nd line here, as only the first line is checked. The same in the other test.

**Review by MalteEbner — APPROVED** [review]
LGTM, one minor question.

**Pull request merged**